kyo-core/shared/src/main/scala/kyo/Option.scala

```diff
@@ -0,0 +1,209 @@
+package kyo
+
+import Option.internal.*
+
+opaque type Option[+T] >: (None | Some[T]) = None | Some[T]
+
+opaque type Some[+T] = T | SomeNone
+
+object Some:
+    given [T: Flat]: Flat[Some[T]]                                   = Flat.unsafe.bypass
+    given [T, U](using CanEqual[T, U]): CanEqual[Option[T], Some[U]] = CanEqual.derived
+
+    def apply[T](v: T): Some[T] =
+        v match
+            case v: SomeNone => v.nest
+            case v: None     => SomeNone()
+            case v           => v
+
+    // TODO avoid allocation
+    def unapply[T](opt: Option[T]): scala.Option[T] =
+        opt.toScala
+
+    extension [T](self: Some[T])
```

Comment:
I had to add methods to `Some` and `None` because Scala isn't able to locate the `Option` extensions for them in some cases.